module-services/service-cellular/src/volte/ImsiParserUS.cpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "ImsiParserUS.hpp"
+#include <log/log.hpp>
+#include <array>
+#include <algorithm>
+#include <stdexcept>
+
+namespace cellular::service
+{
+    constexpr auto usMccCount = 7;
+    const std::array<std::string, usMccCount> usMcc{"310", "311", "312", "313", "314", "315", "316"};
```

Comment:
maybe it would be better to put this in some json config file? it would be more manageable and easy to modify in target devices then

Reply:
Probably it will be implemented as database - GSM standard for IMSI is inconsistent as usual - MNC ( network code ) can be two or three digits, unique ID can be nine or ten digits. It is even allowed to be variable in some countries. Now it's basic implementation to match TMobile.

Reply:
@SP2FET A database will be necessary because of probable future combinations like: _allow (all for country Poland, but not operator Orange) or (any T-Mobile from Asia)_ etc. - making it with `if`s would be a nightmare.